--- python-lxml.changes
+++ saltbundlepy-lxml.changes
@@ -1,63 +1,46 @@
 -------------------------------------------------------------------
-Sat Jul  9 14:10:48 UTC 2022 - Arun Persaud <arun@gmx.de>
-
-- update to version 4.9.1:
-  * Bugs fixed
-    + A crash was resolved when using iterwalk() (or canonicalize())
-      after parsing certain incorrect input. Note that iterwalk() can
-      crash on valid input parsed with the same parser after failing
-      to parse the incorrect input.
-
--------------------------------------------------------------------
-Tue Jun  7 21:14:47 UTC 2022 - Dirk Müller <dmueller@suse.com>
-
-- update to 4.9.0:
-  * The mixin inheritance order in ``lxml.html`` was corrected.
-  * Built with Cython 0.29.30 to adapt to changes in Python 3.11 and 3.12.
-
--------------------------------------------------------------------
-Fri Feb 18 00:36:08 UTC 2022 - Dirk Müller <dmueller@suse.com>
-
-- update to 4.8.0:
-  * GH#337: Path-like objects are now supported throughout the API instead of
-    just strings.
-  * The ``ElementMaker`` now supports ``QName`` values as tags, which always
-    override the default namespace of the factory.
-  * GH#338: In lxml.objectify, the XSI float annotation "nan" and "inf" were spelled in
-    lower case, whereas XML Schema datatypes define them as "NaN" and "INF" respectively.
-  * Built with Cython 0.29.28. 
-
--------------------------------------------------------------------
-Mon Jan 10 23:09:05 UTC 2022 - Dirk Müller <dmueller@suse.com>
-
-- update to 4.7.1:
-  * Chunked Unicode string parsing via ``parser.feed()`` now encodes the input data
-    to the native UTF-8 encoding directly, instead of going through ``Py_UNICODE`` /
-    ``wchar_t`` encoding first, which previously required duplicate recoding in most cases.
-  * The standard namespace prefixes were mishandled during "C14N2" serialisation on Python 3.
-  * ``lxml.objectify`` previously accepted non-XML numbers with underscores (like "1_000")
-    as integers or float values in Python 3.6 and later. It now adheres to the number
-    format of the XML spec again.
-  * LP#1939031: Static wheels of lxml now contain the header files of zlib and libiconv
-    (in addition to the already provided headers of libxml2/libxslt/libexslt).
-  * Wheels include libxml2 2.9.12+ and libxslt 1.1.34 (also on Windows).
-
--------------------------------------------------------------------
-Tue Jan  4 16:03:54 UTC 2022 - Dirk Müller <dmueller@suse.com>
-
-- update to 4.6.5 (bsc#1193752, CVE-2021-43818):
-  * A vulnerability (GHSL-2021-1038) in the HTML cleaner allowed sneaking script
-    content through SVG images.
-  * A vulnerability (GHSL-2021-1037) in the HTML cleaner allowed sneaking script
-    content through CSS imports and other crafted constructs.
-
--------------------------------------------------------------------
-Wed Nov 24 21:23:34 UTC 2021 - Dirk Müller <dmueller@suse.com>
-
-- update to 4.6.4:
-  * A new property ``system_url`` was added to DTD entities.
-  * The ``STATIC_*`` variables in ``setup.py`` can now be passed via env vars.
-- remove python-lxml-test_etree.patch (upstream)
+Mon Apr  4 12:42:21 UTC 2022 - Victor Zhestkov <victor.zhestkov@suse.com>
+
+- Strictly require Python 3.10 with saltbundlepy requrement
+
+-------------------------------------------------------------------
+Fri Apr  1 07:25:51 UTC 2022 - Victor Zhestkov <victor.zhestkov@suse.com>
+
+- With the new update to 4.7.1, the old Bugzilla entries are also
+  fixed:
+  - bsc#1118088 (related to CVE-2018-19787)
+  - bsc#1184177 (related to CVE-2021-28957)
+
+- Update to 4.7.1 (officially released 2021-12-13)
+  Features added
+  - Chunked Unicode string parsing via parser.feed() now encodes the input
+    data to the native UTF-8 encoding directly, instead of going through
+    Py_UNICODE / wchar_t encoding first, which previously required duplicate
+    recoding in most cases.
+
+  Bugs fixed
+  - The standard namespace prefixes were mishandled during "C14N2" 
+serialisation
+    on Python 3.
+    See 
+https://mail.python.org/archives/list/lxml@python.org/thread/
+6ZFBHFOVHOS5GFDOAMPCT6HM5HZPWQ4Q/
+  - lxml.objectify previously accepted non-XML numbers with underscores
+    (like "1_000") as integers or float values in Python 3.6 and later.
+    It now adheres to the number format of the XML spec again.
+  - LP#1939031: Static wheels of lxml now contain the header files of zlib
+    and libiconv (in addition to the already provided headers of
+    libxml2/libxslt/libexslt).
+
+- PDF documentation removed
+
+- Added:
+  * lxml-4.7.1.tar.gz
+
+- Removed:
+  * lxml-4.6.3.tar.gz
+  * lxmldoc-4.5.2.pdf
+  * python-lxml-test_etree.patch
 
 -------------------------------------------------------------------
 Tue Jun  1 19:15:10 UTC 2021 - Matej Cepl <mcepl@suse.com>
