--- python-paramiko.changes
+++ saltbundlepy-paramiko.changes
@@ -1,9 +1,22 @@
 -------------------------------------------------------------------
-Wed Apr  6 06:43:27 UTC 2022 - Steve Kowalik <steven.kowalik@suse.com>
-
-- Add CVE-2022-24302-race-condition.patch:
-  * Fix a race condition between creation and chmod when writing private
-    keys. (bsc#1197279) 
+Mon Apr  4 11:43:53 UTC 2022 - Victor Zhestkov <victor.zhestkov@suse.com>
+
+- Strictly require Python 3.10 with saltbundlepy requrement
+- Fix using deprecated and removed class from collections
+
+- Added:
+  * use-MutableMapping-from-collections.abc.patch
+
+-------------------------------------------------------------------
+Thu Mar 31 08:49:54 UTC 2022 - Victor Zhestkov <victor.zhestkov@suse.com>
+
+- Add patches to ensure that we create new key files directly
+  with a user-only mode, instead of opening, writing, then chmodding.
+  (bsc#1197279) (CVE-2022-24302)
+
+- Added:
+  * Fix-CVE-re-PKey.write_private_key-chmod-race.patch
+  * Use-args-not-kwargs-to-retain-py2-compat-for-now.patch
 
 -------------------------------------------------------------------
 Thu Apr 12 05:33:09 UTC 2021 - Steve Kowalik <steven.kowalik@suse.com>
