--- python-appdirs.spec
+++ saltbundlepy-appdirs.spec
@@ -1,7 +1,7 @@
 #
-# spec file for package python-appdirs
+# spec file for package saltbundlepy-appdirs
 #
-# Copyright (c) 2017 SUSE LINUX GmbH, Nuernberg, Germany.
+# Copyright (c) 2021 SUSE LLC
 #
 # All modifications and additions to the file contributed by third parties
 # remain the property of their copyright owners, unless otherwise agreed
@@ -12,21 +12,28 @@
 # license that conforms to the Open Source Definition (Version 1.9)
 # published by the Open Source Initiative.
 
-# Please submit bugfixes or comments via http://bugs.opensuse.org/
+# Please submit bugfixes or comments via https://bugs.opensuse.org/
 #
 
 
-%{?!python_module:%define python_module() python-%{**} python3-%{**}}
-Name:           python-appdirs
+%{?!saltbundlepy_module:%define saltbundlepy_module() saltbundlepy-%{**}}
+%define pythons saltbundlepy
+
+# Disable python bytecompile for all distros
+# It's called explicitly in the spec
+%global __brp_python_bytecompile %{nil}
+
+Name:           saltbundlepy-appdirs
 Version:        1.4.3
 Release:        0
 Summary:        A small Python module for determining platform-specific dirs
 License:        MIT
 Group:          Development/Languages/Python
-Url:            http://github.com/ActiveState/appdirs
+URL:            http://github.com/ActiveState/appdirs
 Source:         https://files.pythonhosted.org/packages/source/a/appdirs/appdirs-%{version}.tar.gz
-BuildRequires:  %{python_module base}
-BuildRequires:  python-rpm-macros
+BuildRequires:  %{saltbundlepy_module base >= 3.10}
+BuildRequires:  saltbundlepy-rpm-macros
+Requires:       saltbundlepy-base
 BuildRoot:      %{_tmppath}/%{name}-%{version}-build
 BuildArch:      noarch
 %python_subpackages
