--- python-libvirt-python.spec
+++ saltbundlepy-libvirt.spec
@@ -1,7 +1,7 @@
 #
-# spec file for package python-libvirt-python
+# spec file for package saltbundlepy-libvirt
 #
-# Copyright (c) 2022 SUSE LLC
+# Copyright (c) 2021 SUSE LLC
 #
 # All modifications and additions to the file contributed by third parties
 # remain the property of their copyright owners, unless otherwise agreed
@@ -16,31 +16,35 @@
 #
 
 
-# No longer build for python2. Support was dropped upstream in the 6.0.0 release
-%define skip_python2  1
+%{?!saltbundlepy_module:%define saltbundlepy_module() saltbundlepy-%{**}}
+%define pythons saltbundlepy
 
-%{?!python_module:%define python_module() python3-%{**}}
+# Disable python bytecompile for all distros
+# It's called explicitly in the spec
+%global __brp_python_bytecompile %{nil}
+
 %define srcname libvirt-python
-Name:           python-libvirt-python
+
+Name:           saltbundlepy-libvirt
 URL:            https://libvirt.org/
-Version:        8.5.0
+Version:        8.0.0
 Release:        0
 Summary:        Library providing a virtualization API
 License:        LGPL-2.1-or-later
 Group:          Development/Languages/Python
 Source0:        %{srcname}-%{version}.tar.gz
 BuildRequires:  fdupes
-BuildRequires:  libvirt-devel = %{version}
-BuildRequires:  python-rpm-macros
+BuildRequires:  pkgconfig
+%if "%_vendor" == "debbuild"
+BuildRequires:  libvirt-dev
+%else
+BuildRequires:  libvirt-devel
+%endif
+BuildRequires:  saltbundlepy-rpm-macros
 BuildRoot:      %{_tmppath}/%{name}-%{version}-build
-BuildRequires:  %{python_module devel}
-BuildRequires:  %{python_module lxml}
-BuildRequires:  %{python_module pytest}
-%ifpython2
-Provides:       libvirt-python = %{version}
-Obsoletes:      libvirt-python < %{version}
-%endif
-
+BuildRequires:  %{saltbundlepy_module devel >= 3.10}
+BuildRequires:  %{saltbundlepy_module pytest}
+BuildRequires:  %{saltbundlepy_module xml}
 %python_subpackages
 
 %description
@@ -66,7 +70,7 @@
 %python_expand %fdupes %{buildroot}%{$python_sitearch}
 
 %check
-%pytest_arch
+%{pytest_arch -k 'not testScreenshot'}
 
 %files %{python_files}
 %doc README COPYING COPYING.LESSER examples/
