--- python-pycurl.spec
+++ saltbundlepy-pycurl.spec
@@ -1,7 +1,7 @@
 #
-# spec file for package python-pycurl
+# spec file for package saltbundlepy-pycurl
 #
-# Copyright (c) 2019 SUSE LINUX GmbH, Nuernberg, Germany.
+# Copyright (c) 2021 SUSE LLC
 #
 # All modifications and additions to the file contributed by third parties
 # remain the property of their copyright owners, unless otherwise agreed
@@ -16,8 +16,12 @@
 #
 
 
-%define oldpython python
-%{?!python_module:%define python_module() python-%{**} python3-%{**}}
+%{?!saltbundlepy_module:%define saltbundlepy_module() saltbundlepy-%{**}}
+%define pythons saltbundlepy
+
+# Disable python bytecompile for all distros
+# It's called explicitly in the spec
+%global __brp_python_bytecompile %{nil}
 
 %global flavor @BUILD_FLAVOR@%{nil}
 %if "%{flavor}" == "test"
@@ -27,7 +31,8 @@
 %define psuffix %{nil}
 %bcond_with test
 %endif
-Name:           python-pycurl%{psuffix}
+
+Name:           saltbundlepy-pycurl%{psuffix}
 Version:        7.43.0.2
 Release:        0
 Summary:        PycURL -- cURL library module
@@ -42,21 +47,22 @@
 # PATCH-FIX-OPENSUSE python-pycurl-7.43.0-tls-backend.patch -- do not run runtime tests to compare linked libs
 Patch2:         python-pycurl-7.43.0-tls-backend.patch
 Patch3:         disable_randomly_failing_tests.patch
-BuildRequires:  %{python_module devel}
-BuildRequires:  %{python_module setuptools}
+BuildRequires:  %{saltbundlepy_module devel >= 3.10}
+BuildRequires:  %{saltbundlepy_module setuptools}
 BuildRequires:  fdupes
 BuildRequires:  libcurl-devel >= 7.19.0
+%if 0%{?suse_version} == 1315
+BuildRequires:  saltbundle-openssl-devel
+%else
 BuildRequires:  openssl-devel
-BuildRequires:  python-rpm-macros
+%endif
+BuildRequires:  saltbundlepy-rpm-macros
 %if %{with test}
-BuildRequires:  %{python_module bottle}
-BuildRequires:  %{python_module flaky}
-BuildRequires:  %{python_module nose}
+BuildRequires:  %{saltbundlepy_module bottle}
+BuildRequires:  %{saltbundlepy_module flaky}
+BuildRequires:  %{saltbundlepy_module nose}
 %endif
-%ifpython2
-Provides:       %{oldpython}-curl = %{version}
-Obsoletes:      %{oldpython}-curl < %{version}
-%endif
+Requires:       saltbundlepy-base
 %python_subpackages
 
 %description
