--- python-pyinotify.changes
+++ saltbundlepy-pyinotify.changes
@@ -1,3 +1,8 @@
+-------------------------------------------------------------------
+Mon Apr  4 12:55:22 UTC 2022 - Victor Zhestkov <victor.zhestkov@suse.com>
+
+- Strictly require Python 3.10 with saltbundlepy requrement
+
 -------------------------------------------------------------------
 Thu Oct 11 13:04:04 UTC 2018 - jgonzalez@suse.com
 
