--- python-Cython.spec
+++ saltbundlepy-cython.spec
@@ -1,7 +1,7 @@
 #
-# spec file for package python-Cython
+# spec file for package saltbundlepy-cython
 #
-# Copyright (c) 2022 SUSE LLC
+# Copyright (c) 2021 SUSE LLC
 #
 # All modifications and additions to the file contributed by third parties
 # remain the property of their copyright owners, unless otherwise agreed
@@ -16,30 +16,28 @@
 #
 
 
-%{?!python_module:%define python_module() python-%{**} python3-%{**}}
-%define         oldpython python
-%bcond_with test
-Name:           python-Cython
-Version:        0.29.30
+%{?!saltbundlepy_module:%define saltbundlepy_module() saltbundlepy-%{**}}
+%define pythons saltbundlepy
+
+# Disable python bytecompile for all distros
+# It's called explicitly in the spec
+%global __brp_python_bytecompile %{nil}
+
+Name:           saltbundlepy-cython
+Version:        0.29.24
 Release:        0
 Summary:        The Cython compiler for writing C extensions for the Python language
 License:        Apache-2.0
-URL:            https://cython.org/
+URL:            http://www.cython.org
 Source:         https://files.pythonhosted.org/packages/source/C/Cython/Cython-%{version}.tar.gz
 Source1:        python-Cython-rpmlintrc
-BuildRequires:  %{python_module devel}
-BuildRequires:  %{python_module xml}
+BuildRequires:  %{saltbundlepy_module devel >= 3.10}
 BuildRequires:  fdupes
 BuildRequires:  gcc-c++
-BuildRequires:  python-rpm-macros
-Requires:       python-devel
-Requires:       python-xml
+BuildRequires:  saltbundlepy-rpm-macros
+Requires:       saltbundlepy-devel
 Requires(post): update-alternatives
 Requires(postun):update-alternatives
-%ifpython2
-Provides:       %{oldpython}-cython = %{version}
-Obsoletes:      %{oldpython}-cython < %{version}
-%endif
 %python_subpackages
 
 %description
@@ -80,13 +78,6 @@
 %fdupes %{buildroot}%{$python_sitearch}
 }
 
-%check
-%if %{with test}
-%{python_expand export PYTHONPATH=%{buildroot}%{$python_sitearch} PYTHONDONTWRITEBYTECODE=1
-$python runtests.py -v
-}
-%endif
-
 %post
 %python_install_alternative cython cythonize cygdb
 
