--- python-six.spec
+++ saltbundlepy-six.spec
@@ -1,7 +1,7 @@
 #
-# spec file for package python-six
+# spec file for package saltbundlepy-six
 #
-# Copyright (c) 2020 SUSE LLC
+# Copyright (c) 2021 SUSE LLC
 #
 # All modifications and additions to the file contributed by third parties
 # remain the property of their copyright owners, unless otherwise agreed
@@ -16,7 +16,13 @@
 #
 
 
-%{?!python_module:%define python_module() python-%{**} python3-%{**}}
+%{?!saltbundlepy_module:%define saltbundlepy_module() saltbundlepy-%{**}}
+%define pythons saltbundlepy
+
+# Disable python bytecompile for all distros
+# It's called explicitly in the spec
+%global __brp_python_bytecompile %{nil}
+
 # This is not only because of dependency of testsuite, but mostly
 # because of cyclical dependencies between six and Sphinx.
 %global flavor @BUILD_FLAVOR@%{nil}
@@ -27,7 +33,8 @@
 %bcond_with test
 %define psuffix %{nil}
 %endif
-Name:           python-six%{psuffix}
+
+Name:           saltbundlepy-six%{psuffix}
 Version:        1.14.0
 Release:        0
 Summary:        Python 2 and 3 compatibility utilities
@@ -35,15 +42,16 @@
 Group:          Development/Libraries/Python
 URL:            http://pypi.python.org/pypi/six/
 Source:         https://files.pythonhosted.org/packages/source/s/six/six-%{version}.tar.gz
-BuildRequires:  %{python_module base}
+BuildRequires:  %{saltbundlepy_module base >= 3.10}
 BuildRequires:  fdupes
-BuildRequires:  python-rpm-macros
+BuildRequires:  saltbundlepy-rpm-macros
+Requires:       saltbundlepy-base
 BuildArch:      noarch
 %if %{with test}
-BuildRequires:  %{python_module dbm}
-BuildRequires:  %{python_module pytest}
-BuildRequires:  %{python_module py}
-BuildRequires:  python3-Sphinx
+BuildRequires:  %{saltbundlepy_module dbm}
+BuildRequires:  %{saltbundlepy_module pytest}
+BuildRequires:  %{saltbundlepy_module py}
+BuildRequires:  %{saltbundlepy_module sphinx}
 %endif
 %python_subpackages
 
@@ -54,12 +62,12 @@
 both Python versions. See the documentation for more information on
 what is provided.
 
-%package -n python-six-doc
+%package -n saltbundlepy-six-doc
 Summary:        Documentation files for %{name}
 Group:          Documentation/HTML
 Provides:       %{python_module six-doc = %{version}}
 
-%description -n python-six-doc
+%description -n saltbundlepy-six-doc
 Six is a Python 2 and 3 compatibility library. It provides utility
 functions for smoothing over the differences between the Python
 versions with the goal of writing Python code that is compatible on
@@ -103,10 +111,10 @@
 %license LICENSE
 %doc README.rst CHANGES
 %{python_sitelib}/six.py*
-%pycache_only %{python3_sitelib}/__pycache__/*
+%pycache_only %{python_sitelib}/__pycache__/*
 %{python_sitelib}/six-%{version}-py*.egg-info
 %else
-%files -n python-six-doc
+%files -n saltbundlepy-six-doc
 %license LICENSE
 %doc documentation/_build/html
 %endif
